traitlets/traitlets.py

```diff
@@ -644,6 +648,17 @@ def hold(*a):
                 for v in cache.values():
                     self._notify_trait(*v)
 
+    @contextlib.contextmanager
+    def _allow_write(self):
```

Comment:
What's the utility of this context manager? If it's read-only, shouldn't it be read-only?